tests/lib/testing_classes.py

```diff
@@ -340,19 +340,22 @@ def publish_finished(self, dbcon, launched_app, download_test_data,
         timeout = timeout or self.TIMEOUT
         timeout = float(timeout)
 
+        output = ""
         while launched_app.poll() is None:
-            out, err = launched_app.communicate()
-            time.sleep(0.5)
-            if time.time() - time_start > timeout:
-                launched_app.terminate()
-                raise ValueError("Timeout reached")
+            try:
+                out, err = launched_app.communicate(timeout=0.5)
```

Comment:
Shouldn't this `out` also get added to `output`?

Same goes for `err` right - we might need build up the output of that.

Looking at what we're doing - which is waiting around getting all communication. Aren't we better off using something like `subprocess.check_output` or `run` or `call` with the `timeout` argument set?

See [`subprocess.check_output`](https://docs.python.org/3/library/subprocess.html#subprocess.check_output)

Reply:
i believe here we already have a `subprocess` spawned somewhere else, with `subprocess.check_output` we'd be creating a new one, no?

Reply:
You are entirely right - I thought we had full control on what we were to launch but this is indeed using the already launched application.

Reply:
@BigRoy what do you suggest here?

Reply:
Building up the errors similar to output now.